### PR TITLE
Remove support create table with location in Delta Lake connector

### PR DESCRIPTION
**Pull request by marcinsbd**
## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Delta Lake
* Remove support for creating tables with the existing location.
```

### CODE REVIEW

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java [hunk 1]
```diff
@@ -956,6 +956,21 @@ public void testSettingChangeDataFeedEnabledProperty()
...
+    @Test
+    public void testCreateTableWithExistingLocation()
+    {
+        String tableName = "test_legacy_create_table_" + randomNameSuffix();
+
+        assertQuerySucceeds("CREATE TABLE " + tableName + " AS SELECT 1 as a, 'INDIA' as b, true as c");
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 'INDIA', true)");
+
+        String tableLocation = (String) computeScalar("SELECT DISTINCT regexp_replace(\"$path\", '/[^/]*$', '') FROM " + tableName);
+        assertUpdate("DROP TABLE " + tableName);
```
Comment (krvikash) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java:968]: `DROP TABLE` will delete the data as well. You need to use `unregister_table` procedure here.

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 2]
```diff
@@ -935,4 +935,4 @@ connector.
         property to ``false`` to disable the optimized parquet reader by default
         for structural data types. The equivalent catalog session property is
         ``parquet_optimized_nested_reader_enabled``.
-      - ``true``
\ No newline at end of file
+      - ``true``
```
[the remark this answers is not in the document]
Comment (ebyhr) [reply docs/src/main/sphinx/connector/delta-lake.rst:934]: The end of the line was inserted.

[the remark this answers is not in the document]
Comment (marcinsbd) [reply docs/src/main/sphinx/connector/delta-lake.rst:934]: I just learned that my IDE adds an empty line at the end file on save automatically.
Is it ok or should I remove it?  Airlift codestyle seems to be ok with it.

**Comment by Thelin90** [comment]
@marcinsbd 

May I ask why you want to delete this?

We use this heavily to create tables written by spark, delta-rs.

It will create a lot of pain to always have to align schema first in galaxy before doing a spark job. I am sure a lot of people must face this issue?

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 3]
```diff
@@ -393,12 +393,8 @@ ignored. The table schema is read from the transaction log, instead. If the
 schema is changed by an external system, Trino automatically uses the new
```
Comment (krvikash) [inline docs/src/main/sphinx/connector/delta-lake.rst:393]: This section needs to be updated as well.

```
When Delta tables exist in storage, but not in the metastore, Trino can be used
to register them::
  CREATE TABLE example.default.example_table (
    dummy bigint
  )
  WITH (
    location = '...'
  )
Columns listed in the DDL, such as ``dummy`` in the preceeding example, are
ignored. The table schema is read from the transaction log, instead. If the
schema is changed by an external system, Trino automatically uses the new
schema.
```

**Comment by ebyhr** [comment]
Please rebase on master to resolve conflicts.

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeConfig.java [hunk 4]
```diff
@@ -449,21 +448,6 @@ public DeltaLakeConfig setUniqueTableLocation(boolean uniqueTableLocation)
         return this;
     }
 
-    @Deprecated
-    public boolean isLegacyCreateTableWithExistingLocationEnabled()
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeConfig.java:453]: Move to `@DefunctConfig`.

**Review by ebyhr — APPROVED** [review]

**Review by krvikash — APPROVED** [review]

**Comment by mosabua** [comment]
Does that mean that create table with external location is not remove completely in favour of the register table procedure? And this was the last part of that removal? @marcinsbd @ebyhr

**Comment by ebyhr** [comment]
@Thelin90 Since the specified location holds the table definitions, passing dummy columns doesn't make sense. We removed the feature to reduce the maintenance cost.

@mosabua Yes, we deprecated `delta.legacy-create-table-with-existing-location.enabled` few month ago and this PR removed the feature completely.

**Pull request merged**